### PR TITLE
Add response

### PR DESCRIPTION
**Pull request by khan-ngo**
Comfort: 3
Clarity: 3

I used Google and my class notes to help with this diagnostic.

### CODE REVIEW

**Comment by khan-ngo** [comment]
I didn't mean to push from my master branch ... I forgot to checkout to response branch before working on diagnostic.md

File: diagnostic.md [hunk 1]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
 What is the purpose of a backend?
 
 ```md
-// your response here
+The back-end is where things happens behind the scenes ... it is were the processing occurs ..
```
Comment (jordanallain) [inline diagnostic.md:8]: Most importantly it is where our data gets stored, that ensures we are able to save data that will persist between sessions and different users.

File: diagnostic.md [hunk 2]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
...
 ```
 
 Which layer in the MVC pattern is used by the controller to fetch data?
 
 ```md
-// your response here
+the controller is the "C" in the model
```
Comment (jordanallain) [inline diagnostic.md:14]: In MVC M stands for model and C stands for controller. For this question we are asking which layer does the controller use to fetch data. The correct answer is model.

File: diagnostic.md [hunk 3]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
...
 ```md
-// your response here
+Create, Read, Update, Delete
 ```
 
 In which part of the MVC pattern can we find C.R.U.D actions?
 
 ```md
-// your response here
+the CRUD occurs at the data level - our Model
```
Comment (jordanallain) [inline diagnostic.md:38]: Whoops, should be controller.

File: diagnostic.md [hunk 4]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
...
 
 A user action fires a `GET` request for `/people/1`. Explain in detail each step
 required for data to be returned to the client. (bullet points or ordered list)
 
 ```md
-// your response here
+GET access the API,
+Routes to certain action
+Action defines what needs to be done
+View returns the ooutput
```
Comment (jordanallain) [inline diagnostic.md:58]: So we won't be using views on our back end. Instead the router chooses the controller, the controller tells the model what to get from the database, the model gets the data and gives it to the controller to serve to the client.

File: diagnostic.md [hunk 5]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
...
 ```
 
 What is the command to generate a new rails-api app?
 
 ```bash
-// your response here
+bundle install in the rails api repo ;-)
+it works ... I didn\'t ask why ...
```
Comment (jordanallain) [inline diagnostic.md:65]: To generate a new rails app we would use this command:

`rails new <app_name>`

File: diagnostic.md [hunk 6]
```diff
@@ -5,74 +5,89 @@ Place your responses inside the fenced code-blocks where indicated by comments.
...
 
 What is the command to scaffold a pet with a name and age attributes (hint:
 Also think of the data types for each attribute)?
 
 ```bash
-// your response here
+ def set_name(name, age)
+   @name = name
+   @age = age
+ end
```
Comment (jordanallain) [inline diagnostic.md:92]: So here we wanted the terminal command to scaffold a pet:

`rails generate scaffold pet name:string age:integer`

**Comment by jordanallain** [comment]
6/12

Looks like you struggled a bit with this, review the answers I shared with you and reach out if you need some help with anything!

🇯🇴 🇯🇴 🇯🇴 🇯🇴 🇯🇴 🇯🇴

**Pull request closed without merging**